### PR TITLE
Dmp 2662.database updates

### PR DESCRIPTION
**Pull request by cmagowan-hmcts**
https://tools.hmcts.net/jira/browse/DMP-2662

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/common/entity/ExternalObjectDirectoryEntity.java [hunk 1]
```diff
@@ -87,6 +87,9 @@ public class ExternalObjectDirectoryEntity extends CreatedModifiedBaseEntity imp
     @Column(name = "is_response_cleaned")
     private boolean responseCleaned;
 
+    @Column(name = "update_retention")
+    private boolean updateRetention;
```
Comment (johnrace0) [inline src/main/java/uk/gov/hmcts/darts/common/entity/ExternalObjectDirectoryEntity.java:91]: some entities are using Boolean - others boolean - do we know which one is preferred or whether it matters?

File: src/main/java/uk/gov/hmcts/darts/common/entity/CourtCaseEntity.java [hunk 2]
```diff
@@ -86,6 +95,16 @@ public class CourtCaseEntity extends CreatedModifiedBaseEntity {
...
 
+    @Column(name = IS_DELETED)
+    private boolean isDeleted;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "deleted_by")
+    private UserAccountEntity deletedBy;
+
+    @Column(name = "deleted_ts")
+    private OffsetDateTime deletedTs;
```
Comment (johnrace0) [inline src/main/java/uk/gov/hmcts/darts/common/entity/CourtCaseEntity.java:106]: another consistency question - some fields are "Ts", some "Timestamp", some "DateTime"

File: src/main/java/uk/gov/hmcts/darts/common/entity/CaseDocumentEntity.java [hunk 3]
```diff
@@ -43,17 +48,37 @@ public class CaseDocumentEntity {
...
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "marked_for_manual_del_by")
+    private UserAccountEntity markedForManualDelBy;
+
+    @Column(name = "marked_for_manual_del_ts")
+    private OffsetDateTime markedForManualDelTs;
 
-    @Column(name = "uploaded_ts", nullable = false)
-    private OffsetDateTime uploadedTs;
+    @Column(name = "retain_until_ts")
```
Comment (johnrace0) [inline src/main/java/uk/gov/hmcts/darts/common/entity/CaseDocumentEntity.java:81]: these changes are defined in DMP-2658 - see my comments on there

File: src/main/java/uk/gov/hmcts/darts/common/entity/base/ModifiedBaseEntity.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import java.time.OffsetDateTime;
+
+@MappedSuperclass
+@Getter
+@Setter
+public class ModifiedBaseEntity {
+
+    @UpdateTimestamp
+    @Column(name = "last_modified_ts")
+    private OffsetDateTime lastModifiedDateTime;
```
Comment (johnrace0) [inline src/main/java/uk/gov/hmcts/darts/common/entity/base/ModifiedBaseEntity.java:22]: see other comment on _ts naming consistency

File: src/main/resources/db/migration/common/V1_288_1__v63_v64_v65_database_updates.sql [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+ALTER TABLE case_document ALTER COLUMN checksum DROP NOT NULL;
+ALTER TABLE case_document ALTER COLUMN is_hidden SET DEFAULT false;
+ALTER TABLE case_document ADD COLUMN marked_for_manual_deletion BOOLEAN NOT NULL DEFAULT false;
+ALTER TABLE case_document ADD COLUMN retain_until_ts TIMESTAMP WITH TIME ZONE;
+ALTER TABLE case_document ADD COLUMN last_modified_ts TIMESTAMP WITH TIME ZONE;
+update case_document set last_modified_ts = uploaded_ts;
+ALTER TABLE case_document ALTER COLUMN last_modified_ts SET NOT NULL;
+ALTER TABLE case_document ADD COLUMN last_modified_by INTEGER;
+
+ALTER TABLE court_case ADD COLUMN is_retention_updated BOOLEAN NOT NULL DEFAULT false;
```
Comment (johnrace0) [inline src/main/resources/db/migration/common/V1_288_1__v63_v64_v65_database_updates.sql:76]: will this mean that existing rows will be set to default value?

File: src/main/java/uk/gov/hmcts/darts/common/entity/ObjectHiddenReasonEntity.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+public class ObjectHiddenReasonEntity {
+
+    @Id
+    @Column(name = "ohr_id")
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "ohr_gen")
+    @SequenceGenerator(name = "ohr_gen", sequenceName = "ohr_seq", allocationSize = 1)
+    private Integer id;
+
+    @Column(name = "ohr_reason")
+    private String description;
```
Comment (johnrace0) [inline src/main/java/uk/gov/hmcts/darts/common/entity/ObjectHiddenReasonEntity.java:24]: reason or description?

File: src/main/java/uk/gov/hmcts/darts/common/entity/CaseDocumentEntity.java (hunk 3 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/common/entity/CaseDocumentEntity.java:81]: i mentioned this in the standup this morning and said that all the changes were done and just needed a review and that whoever was picking up 2658 didn't need to do them

File: src/main/resources/db/migration/common/V1_288_1__v63_v64_v65_database_updates.sql (hunk 5 above)
Comment (cmagowan-hmcts) [reply src/main/resources/db/migration/common/V1_288_1__v63_v64_v65_database_updates.sql:76]: yes

File: src/main/java/uk/gov/hmcts/darts/common/entity/CaseDocumentEntity.java (hunk 3 above)
Comment (johnrace0) [reply src/main/java/uk/gov/hmcts/darts/common/entity/CaseDocumentEntity.java:81]: missed that - blaming it on my comms problem!

**Review by johnrace0 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/darts/common/entity/base/ModifiedBaseEntity.java (hunk 4 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/common/entity/base/ModifiedBaseEntity.java]: renamed to timestamp

File: src/main/java/uk/gov/hmcts/darts/common/entity/CourtCaseEntity.java (hunk 2 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/common/entity/CourtCaseEntity.java]: renamed to timestamp

**Pull request merged**